### ISSUE TITLE
System.Windows.Forms dependency added when installing Microsoft.Identity.Client in an ASP.Net project

### ISSUE THREAD

**Issue by ggirard07** [AzureAD/microsoft-authentication-library-for-dotnet#757]
**Is your feature request related to a problem? Please describe.**
Currently installing `Microsoft.Identity.Client` in an ASP.Net project result in a reference to `System.Windows.Forms` added to the web project despite it makes no sense.

Is it possible to conditionally pulls-in this reference only when needed, ie when targeting the Windows platform?

**Describe the solution you'd like**
Not too familiar with NuGet and nuspec. but is it possible to move the `System.Windows.Forms` reference in the nuspec file from the common section to a specific group with a TFM related to the Windows platform?

**Describe alternatives you've considered**
Not sure it is even possible in the first place, or if the reference is actually required.

**Additional context**
This dependency just makes no sense for a web project. I stumbled some times to understand why it appeared in my project.
Aim is just to limit dependencies to what is really required...

**Comment by bgavrilMS** [comment]
Hi @ggirard07 - MSAL indeed takes a dependency on Win Forms, because in client apps (e.g. a .net console app or a WPF app), we need to be able to pop up a web browser hosted in a WinForms panel. 

I can't see an easy way for us to remove this dependency - we'd have to split the nuget package in 2, a base one and one that provides the WinForms based auth. 

While I understand that win forms does not make sense for web project, I fail to see the impact: 

- does this affect your runtime, i.e. is this assembly loaded into memory when you do auth in your asp.net project?
- how does this affect your maintenance?

**Comment by mdesousa** [comment]
This is causing an issue for our project because we are referencing this library from a windows service, and suddenly our code analyzers are warning us that we should flag our application entry points with as an STA thread. We could ignore this warning... but this is very concerning because if WinForms objects happen to be created, we run the risk of the application freezing if it receives windows messages that can't be handled: https://blogs.msdn.microsoft.com/dsui_team/2012/10/31/debugging-windows-forms-application-hangs-during-systemevents-userpreferencechanged/
The bottom line is that this is restricting our ability to deliver robust code... please consider splitting this library.
